super-turbo-ignore.js

```diff
@@ -0,0 +1,185 @@
+// Derived from vercel/turborepo:packages/turbo-ignore/src/ignore.ts@29aeaf0
+// See MIT License from Vercel, Inc
+
+/** @import { CommitCheck, PackageJson, DryRun } from './types' */
+
+const { execSync } = require("node:child_process");
+const fs = require("node:fs");
+
+// the "≫" symbol
+const LOG_PREFIX = "\u226B  ";
+
+function info(message) {
+	process.stderr.write(LOG_PREFIX + message + "\n");
+}
+
+function error(...args) {
+	console.error(LOG_PREFIX, ...args);
+}
+
+function output(...args) {
+	console.log(...args);
+}
+
+/**
+ * Adapted from turbo-ignore/src/checkCommit.ts@257d0c0
+ * @param {string} workspace - the workspace involved
+ * @returns {CommitCheck}
+ */
+function checkCommit(workspace) {
+	const commitMessage = execSync("git show -s --format=%B").toString();
+
+	const forceWorkspaceDeploy = `[vercel deploy ${workspace}]`;
+	if (commitMessage.includes(forceWorkspaceDeploy)) {
+		return {
+			result: "deploy",
+			reason: `Found commit message: ${forceWorkspaceDeploy}`,
+		};
+	}
+
+	return {
+		result: "continue",
+		reason: `No deploy or skip string found in commit message.`,
+	};
+}
+
+/**
+ * Adapted from turbo-ignore/src/getTurboVersion.ts@29aeaf0
+ * @returns {string}
+ */
+function getTurboVersion() {
+	try {
+		const raw = fs.readFileSync("package.json", "utf8");
+		/** @type {PackageJson} */
+		const packageJson = JSON.parse(raw);
+		const dependencies = packageJson.dependencies?.turbo;
+		const devDependencies = packageJson.devDependencies?.turbo;
+		const turboVersion = dependencies || devDependencies;
+		if (turboVersion !== undefined) {
+			info(`Inferred turbo version "${turboVersion}" from "package.json"`);
+			return turboVersion;
+		}
+	} catch (e) {
+		error(`"${packageJsonPath}" could not be read`);
+		throw e;
+	}
+}
+
+/**
+ * Check if the deployment can be skipped by comparing with the previous commit
+ * Allows forcing deployment from commit message
+ * Assumes package has lockfile, package manager is set, and commit has reachable parent
+ * @param {string} workspace - the workspace to use
+ * @param {string} task - the task to run
+ * @returns {boolean} whether or not the deployment can be skipped
+ */
+function superTurboIgnore(task, workspace) {
+	info(`Using Turborepo to determine if this project is affected by the commit...\n`);
+
+	// check for TURBO_FORCE and bail early if it's set
+	if (process.env.TURBO_FORCE === "true") {
+		info("`TURBO_FORCE` detected");
```

Comment:
Matter of preference but might be clearer to say `TURBO_FORCE environment variable detected`

Reply:
Was copied from source.